File: src/main/java/Q10/Q10_03/Main.java
package Q10.Q10_03;

import java.util.Scanner;

public class Main {
    private static int[] dy;

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] arr = new int[n];
        dy = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = in.nextInt();
        }
        Main main = new Main();
        int rs = main.solution(arr);
        System.out.println(rs);

    }

    private int solution(int[] arr) {
        int answer = 0;
        dy[0] = 1;
        for (int i = 1; i < arr.length; i++) {
            int maxLen = 0;
            for (int j = i - 1; j >= 0; j--) {
                if (arr[j] < arr[i] && dy[j] > maxLen) {
                    maxLen = dy[j];
                }
            }
            dy[i] = maxLen + 1;
            answer = Math.max(answer, dy[i]);
        }
        return answer;
    }
}

File: src/main/java/Q10/Q10_03/main.py
n = int(input())
arr = list(map(int, input().split()))

dy = [1]
answer = 0
for i in range(1, len(arr)):
    max_num = 0
    for j in range(i-1, -1,-1):
        if (arr[j] < arr[i]) & (dy[j] > max_num):
            max_num = dy[j]
    dy.insert(i, max_num + 1)
    answer = max(answer, dy[i])

print(answer)
